subsys/mgmt/osdp/src/osdp.c

```diff
@@ -199,7 +204,24 @@ static int osdp_init(const struct device *arg)
 
 	/* setup OSDP */
 	ctx = osdp_build_ctx(&channel);
-	if (osdp_setup(ctx)) {
+	if (ctx == NULL) {
+		LOG_ERR("Failed to build osdp ctx");
+		k_panic();
+	}
+
+#ifdef CONFIG_OSDP_SC_ENABLED
```

Comment:
Better to have `if (IS_ENABLED(CONFIG_OSDP_SC_ENABLED)) {` here if possible

Reply:
Will address this in my next patchset.